### DISCUSSION TITLE
the-best-media-server-cpu-in-the-world/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#230] [Announcements]
# the-best-media-server-cpu-in-the-world/

After a solid 8 months of testing, I think we have a big enough sample size to draw some conclusions. The best media server CPU is...

https://blog.ktz.me/the-best-media-server-cpu-in-the-world/

**Comment by Foxy-NC** [comment]
Very good post here👍
My 2 cents: Igpu is only necessary if you really need to transcode. If loading from LAN, any media box will decode HEVC just fine (not AV-1 yet) and pass-through the audio to a soundbar or Amp (for DD, Atmos, DTS-X...).
It only make sense when loading content from outside of the LAN so your Internet connection doesn't max out your upload speed.
That's why I use an Nvidia TV Shield and don't activate transcode functionality on Plex/Jellyfin. 😉

**Reply by ironicbadger** [reply]
I bet you’d be surprised the amount of things that get transcoded by accident for subtitles or audio. Nvidia Shield is amazing, no discussion there! Mine is from 2015 and still going strong. Incredible device.

Try this in your Jellyfin under “playback reporting plugin > query”

```SQL
SELECT COUNT(ROWID), PlaybackMethod 
FROM PlaybackActivity 
GROUP BY PlaybackMethod
```

**Comment by fthorsen76** [comment]
I went with an ERYING mobo with an 12700H soldered on from Ali Express. It's seriously fast while not using too much power. It only supports 64GB RAM, but it's enough to run some vm's or many containers. I've added a 10GB NIC and two M2s in RAID for OS and apps, while media resides on a NAS. Very happy with this setup.

**Reply by BrandonG777** [reply]
Just upgraded from LGA2011/X79 to z790 with i5-12500T. I also just upgraded my main desktop from a dual CPU HP z840 workstation to i5-14600K. While the 14600k is a beast, it gets hot.. too hot for my server closest with 3x120mm circulating the air, while the 12500T rarely sees temps above 55C using a 280mm liquid ID cooler. And yes DDR5 or bust from here on out! Well… at least until DDR5/6 becomes more readily available. Also worth noting, I’ve completely offloaded cuda processing (Quadro RTX 4000) to QSV, #winning! Now if only Immich could use QSV or Coral (probably more applicable in this situation). Coral USB working great with Frigate/QSV, overall power reduced by 100watts and a LOT less heat being generated. Spun up a Stable Diffusion container just so the Quadro doesn’t get lonely.

**Reply by BrandonG777** [reply]
Last note, on the m.2 RAID mention, I picked up an Amazon returned gigabyte z790 gaming ax board for a little over $100. This includes 4 m.2 slots. I’m currently using old 4x1TB drives for my cache pool (RAIDZ).

**Reply by ironicbadger** [reply]
These are the ones Jeff on Craft Computing covered? They look pretty nice actually. Is there a catch with them?

**Comment by neobiotics** [comment]
Hey, thanks a lot to you and all other contributors! Awesome work! One question though. Why not 7th gen? The sport the same HD630 iGPU and are a lot cheaper than 8th gen, supposedly because they are not supported by windows 11.

**Reply by ironicbadger** [reply]
Great point, the 7th are a nice sweet spot. In my personal infrastructure I had an i7-8700k in my gaming rig and thought it would make sense to standardise on 8th gen in the house so I could swap chips around at will. 

My blue iris box has 8th gen, so does my backup box, my old Proxmox homelab playground node, and my desktop. I never actually swapped CPUs around like I thought I would but I _could have done_. 

I might modify the text slightly so that it prompts folks to start at 7th not 8th. Codec support jumps with HEVC encode being added from 6th to 7th and from then on out is quite a bit more stable. How's motherboard availability for them?

**Reply by atunguyd** [reply]
That's what i settled on for my media server (linux plex server). An i3 7100t that i picked up second hand for a steal.  Has the qsv support for any re encoding requirements and really low power.  Sitting on linux it has no problems serving up other duties in the house from sonarr/radarr and file server duties

**Reply by neobiotics** [reply]
Motherboards are easily available at least in Germany. An ASUS Prime B250M-C for instance, which comes with 6 SATA ports, four DIMM slots, two M.2 slots and an intel i219V NIC can be acquired for a measly 50€. Pair it with an i5-7400 (25€) and 16 gigs of DDR4 (20-25€) and you have the base for a real budget media server build.

**Comment by cdgriffith** [comment]
A note on the ECC memory, is that DDR5 does not come default with side-band ECC, but on-die ECC. It does not have the additional lanes running to the CPU to detect bad flips.  You can get true ECC DDR5 memory and pair it with a W680 intel board with the chip you listed, but it takes that special chipset and memory.

If the media server is just for playing stuff, doesn't really matter, but if you're hosting data on it, you'd want side-band ECC memory. More info can be found https://codecalamity.com/do-you-really-need-ecc-memory/

**Reply by ironicbadger** [reply]
Facts here! Thanks for a really useful comment.

**Comment by desheh** [comment]
What about AMD CPUs (well, APUs)? How do they fare in this game? Have you tried them? 

You could find a mini pc / nas mono (with a soldered chip) with an AMD 5700u for around 200$ on AliExpress, and they seem quite capable (especially when compared to the N100).

**Reply by neobiotics** [reply]
AMD graphics are not recommended because of their inferior transcoding quality, at least by the jellyfin project.

https://jellyfin.org/docs/general/administration/hardware-selection/

**Reply by cdgriffith** [reply]
Anything AMD with RDNA2 or newer is going to be nearly visually identical compared to intel QSV or NVENC with H.264 encodings if setup properly. https://codecalamity.com/amd-re-introduces-the-b-frame/  

With H.265 (4K / HDR movies) they may still lag behind some in quality. https://codecalamity.com/hardware-encoding-4k-hdr10-videos/ 

Will note with my testing that intel is the only one that would randomly have suddenly low visual quality out of nowhere without warning, so I don't recommend it for storing conversions, but is a great solution for just being a real time transcoder like these media setups.

**Comment by oxbridgepentathlon** [comment]
Very useful.  Much more interesting to see time comparisons rather than the usual comparison of how many simultaneous real-time transcodes can be achieved.  The obvious question is what happens if you throw in an Nvidea card and use nvenc?  Is it would be very interesting to see a bunch of data points for different discrete GPUs to see if they can beat quicksync.

**Reply by neobiotics** [reply]
My guess would be that they are faster while being much less energy eficcient. The thing is, in media server use, the transcode has to be just fast enough and you dont necessarily gain anything from even faster transcodes. I mean, as long as you dont start playing back your movies faster, I guess ;-)
So spending the extra money on the GPU and the electricity doesnt make much sense for a media server.
Nevertheless the comparison would be interesting. Both in terms of transcoding speed and performance per watt.

That all changes of course if you plan on re-ripping your entire media collection or something like that...

**Comment by oxbridgepentathlon** [comment]
Is the HEVC 8bit 4k or 1080p?  The chart doesn't say.

**Comment by jcass8695** [comment]
Thank you for the brilliant post, it's an extremely useful source of information for someone researching a new project.

One question I have is why the first two series in the top graph, `h264_1080p` & `h264_4k` don't specify the bit depth whereas the other series do. Is that a convention or a mistake? Asking for my own understanding, as I learn about these things.
